### PR TITLE
OT-1772 refactor ktcpp focal length 

### PR DESCRIPTION
**Pull request by SergeyKrivohatskiy**

### CODE REVIEW

File: keentools_facebuilder/fbloader.py [hunk 1]
```diff
@@ -456,42 +456,34 @@ def _unfix_not_in_groups(fb, head):
...
                     _unfix_all(fb, head)
-                    mode = 'FB_ESTIMATE_VARYING_FOCAL_LENGTH'
                 elif head.manual_estimation_mode == 'current_estimation':
+                    fb.set_varying_focal_length_estimation()
                     _fix_all_except_this(fb, head, kid)
-                    mode = 'FB_ESTIMATE_VARYING_FOCAL_LENGTH'
                 elif head.manual_estimation_mode == 'same_focus':
-                    _unfix_all(fb, head)
-                    mode = 'FB_ESTIMATE_STATIC_FOCAL_LENGTH'
+                    fb.set_static_focal_length_estimation(settings.get_projection_matrix()[0][0])
```
Comment (SergeyKrivohatskiy) [inline keentools_facebuilder/fbloader.py:481]: settings.get_projection_matrix()[0][0]?

Comment (amilovsky) [reply keentools_facebuilder/fbloader.py:481]: Not sure. Have a look at coords.projection_matrix(). There is a special scale factor dependent on camera orientation (and blender view fixes):

def projection_matrix(w, h, fl, sw, near, far, scale=1.0):
    z_diff = near - far
    fl_to_sw = fl / sw
    return np.array(
        [[scale * w * fl_to_sw, 0, 0, 0],
         [0, scale * w * fl_to_sw, 0, 0],
         [-w / 2, -h / 2, (near + far) / z_diff, -1],
         [0, 0, 2 * near * far / z_diff, 0]]
    ).transpose()

Comment (amilovsky) [reply keentools_facebuilder/fbloader.py:481]: coords.focal_by_projection_matrix(projection_matrix, sensor_width) looks like solving the problem

File: keentools_facebuilder/fbloader.py [hunk 2]
```diff
@@ -456,42 +456,37 @@ def _unfix_not_in_groups(fb, head):
...
                     if camera.is_in_group():
-                        for cam in head.cameras:
-                            fb.set_focal_length_fixed_at(
-                                cam.get_keyframe(),
-                                cam.image_group != camera.image_group
-                                or not cam.auto_focal_estimation)
-                        mode = 'FB_ESTIMATE_STATIC_FOCAL_LENGTH'
+                        sensor_width = head.sensor_width if head.sensor_width != -1 \
+                            else Config.default_sensor_width
+                        proj_mat = settings.get_projection_matrix()
```
Comment (amilovsky) [inline keentools_facebuilder/fbloader.py:464]: I cannot find where settings.get_projection_matrix() method defined.
get_projection_matrix() belongs to camera object, not scene

**Pull request merged**